src/main/java/tetris/Action.java

```diff
@@ -15,11 +15,22 @@ public String toString() {
         public void attempt(AbstractTetromino tetromino, Grid grid) {
             tetromino.rotateRight();
             for (int i = 0; i < 4; i++) {
+                if (!grid.isFree(tetromino.leftCoor())) {
+                    tetromino.moveRight();
+                    Logger.log(this, Logger.LogType.INFO,
+                        "rotated tetromino clockwise and moved to the right to make it possible");
+                }
+            }
+            for (int i = 0; i < 4; i++) {
+                if (!grid.isFree(tetromino.rightCoor())) {
+                    tetromino.moveLeft();
+                    Logger.log(this, Logger.LogType.INFO,
+                        "rotated tetromino clockwise and moved to the left to make it possible");
+                }
+            }
+            for (int i = 0; i < 4; i++) {
                 if (!grid.isFree(tetromino.get(i))) {
                     tetromino.rotateLeft();
-                    Logger.log(this, Logger.LogType.INFO,
-                        "tried to rotate tetromino clockwise but failed");
-                    break;
                 }
             }
             Logger.log(this, Logger.LogType.INFO, "rotated tetromino clockwise");
```

Comment:
Perhaps it's an idea to let abstracttetromino implement iterable and then overload grid.isFree such that it takes a tetromino and performs the iteration in the isFree method. This would also mean making the Action enum a lot more comprehensible.